### PR TITLE
fix: cloudinit changes to allow user to pass custom script

### PR DESCRIPTION
**Pull request by karthicgit**

### CODE REVIEW

File: modules/oke/variables.tf [hunk 1]
```diff
@@ -53,6 +53,10 @@ variable "use_node_pool_volume_encryption" {
 
 variable "node_pool_volume_kms_key_id" {}
 
+variable "node_pool_node_metadata" {}
```
Comment (hyder) [inline modules/oke/variables.tf:56]: This should be a map right. Please ensure you test with multiple node pools

**Review by hyder — CHANGES_REQUESTED** [review]
- [ ] change variable type to map in the submodule
- [ ] add docs in a separate page

File: modules/oke/variables.tf (hunk 1 above)
Comment (karthicgit) [reply modules/oke/variables.tf]: added variable type in submodule.

File: docs/terraformoptions.adoc [hunk 2]
```diff
@@ -698,6 +698,16 @@ node_pools = {
...
 
+|cloudinit_nodepool_common
+|Cloudinit common for all nodepools when no specific script mentioned for nodepool in cloudinit_nodepool.
+|
+|""
+
+|cloudinit_nodepool
+|Cloudinit specific to nodepool to override cloudinit_nodepool_common.
+|
+|{}
```
Comment (hyder) [inline docs/terraformoptions.adoc:709]: Please provide an example

File: docs/cloudinit.adoc [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+/usr/libexec/oci-growfs -y
+
+timedatectl set-timezone ${worker_timezone}
+----
+
+* To customize this you can modify the above script and pass the script as input variable to `cloudinit_nodepool_common`. This script will be used across all the nodepools.
+
+* To use specific cloud init script for a nodepool pass the script as input variable to `cloudinit_nodepool` as a map.This will take precedence over `cloudinit_nodepool_common` for that nodepool.
+Ex: cloudinit_nodepool        = {
+  #np1 = "/tmp/np1cloudinit.sh"
```
Comment (hyder) [inline docs/cloudinit.adoc:32]: Are your filenames just a convention or something that's expected by nodepool name?

File: docs/cloudinit.adoc [hunk 4]
```diff
@@ -0,0 +1,37 @@
+= CloudInit
```
Comment (hyder) [inline docs/cloudinit.adoc:1]: Typo: It's **cloud-init**.

File: docs/cloudinit.adoc (hunk 3 above)
Comment (karthicgit) [reply docs/cloudinit.adoc:32]: the filename can be anything its just an example.

**Review by hyder — APPROVED** [review]
Approved

**Pull request merged**